lib/jekyll/reader.rb

```diff
@@ -131,12 +132,15 @@ def filter_entries(entries, base_directory = nil)
     # subfolder - The String representing the directory to read.
     #
     # Returns the list of entries to process
-    def get_entries(dir, subfolder)
+    def get_entries(dir, subfolder, limit)
       base = site.in_source_dir(dir, subfolder)
       return [] unless File.exist?(base)
 
       entries = Dir.chdir(base) { filter_entries(Dir["**/*"], base) }
       entries.delete_if { |e| File.directory?(site.in_source_dir(base, e)) }
+
+      entries_limit = [entries.length, limit].min
+      limit.positive? ? entries[-entries_limit, entries_limit] : entries
```

Comment:
I believe this line can be refactored to use `:last(n)`:
```suggestion
      limit.positive? ? entries.last(entries_limit) : entries
```
But, do benchmark the two routes and see if there's a benefit from making the change..